### ISSUE TITLE
⌘-clicking on path:line:col in iTerm2 opens a new window and ignores :line:col.

### ISSUE THREAD

**Issue by marcelocantos** [Microsoft/vscode#22845]
- VSCode Version: 1.10.2 8076a19fdcab7e1fc1707952d652f0bb6c6db331
- OS Version: macOS Sierra 10.12.3 (16D32)

Steps to Reproduce:

1. Generate some compiler output in iTerm that includes _path:line:col_ entries, e.g. `foo/bar/file.go:123:25`.
2. Ensure the file is opened in a VSCode tab. For good measure, have another tab in focus.
3. ⌘-click on the file in iTerm2.
  - The file opens in a new VSCode window with the cursor positioned at the start of the file.
4. Close the new VSCode window.
5. Run `code -g foo/bar/file.go:123`.
  - The existing VSCode tab moves comes into focus with the specified line centered, and the cursor is positioned at the specified line and column. In short, behavior is as expected.
6. Repeating steps 1–5 with just _path:line_ (no _:col_) exhibits the same anomalous behavior.

I switched away from Sublime Text 3 yesterday, and it was behaving as expected, so I suspect it's a problem with how VSCode specifies file opening behavior, not with iTerm2, _per se_.

**Comment by bpasero** [comment]
@marcelocantos to be clear, you need to set the -g option to open a file on a specific line/col. Otherwise it will not work.

**Comment by marcelocantos** [comment]
@bpasero The problem is that I can't control the way ⌘-click works in iTerm2. It invokes the associated app (_Visual Studio Code.app_, not _code_) and seems to use some kind of behind the scenes protocol to tell the app which line and column to place the cursor on.

I tested this by creating a proxy app that invokes the real app using code -g and then registering it as the app for opening .go files. The proxy only gets given the path on the command line, not the line or column number. I checked environment variables too, and there's no hint of line:col information there either.

Whatever the approach is, it works out of the box with Sublime and Atom.
